collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -595,12 +598,23 @@ protected void onActivityResult(int requestCode, int resultCode,
             return;
         }
 
-        switch (requestCode) {
-            case BARCODE_CAPTURE:
+        // For handling results returned by the Zxing Barcode scanning library
+        IntentResult barcodeScannerResult = IntentIntegrator.parseActivityResult(requestCode, resultCode, intent);
+        if (barcodeScannerResult != null) {
+            if (barcodeScannerResult.getContents() == null) {
+                // request was canceled...
+                Timber.i("QR code scanning cancelled");
+            } else {
                 String sb = intent.getStringExtra("SCAN_RESULT");
                 ((ODKView) currentView).setBinaryData(sb);
                 saveAnswersForCurrentScreen(DO_NOT_EVALUATE_CONSTRAINTS);
-                break;
+                return;
```

Comment:
This statement makes the `refreshCurrentView()` line at the end of this method ([line 761](https://github.com/MukundAnanthu/collect/blob/27a051c1bf64ea38a0b001b6e2eace8c7c378ab7/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java#L761)) unreachable.
So you should call this function `refreshCurrentView()` before returning from this method.